sonic_sku_create/sonic_sku_create.py

```diff
@@ -0,0 +1,329 @@
+		
+		
+	def msn2700_specific(self) :
+		for fp, values in self.fpp_split.items():
+			splt_arr = sorted(values[0])
+			idx_arr = sorted(values[1])
+			
+			splt = len(splt_arr)
+			fp=int(fp)
+			if ((fp%2) == 1 and splt == 4) :
+				try :
+					self.portconfig_dict.pop(str(int(idx_arr[-1])+1) )
+				except KeyError:
+					continue
+				print ("MSN2700 -  Disabled front panel port ",fp+1, "due to port ",fp,"Split by 4") 
+			elif ((fp%2) == 0 and splt == 4) :
+				for idx in idx_arr :
+					self.portconfig_dict.pop(idx,None )
+				print ("MSN2700 -  even front panel ports (",fp,") are not allowed to split by 4")
+				return -1
+
+
+	def l2_mode(self,l2_sku) :
+		if not os.path.exists(DEFAULT_DEV_PATH+self.platform+"/"+l2_sku):
+			print ("Error - path:", l2_sku, " doesn't exist")
+		print( "Starting L2 mode configuration based on "+l2_sku+" SKU")	
+		cfg_db_str = "\n{\n\t\"DEVICE_METADATA\": {\n\t\t\"localhost\": {\n\t\t\t\"hostname\": \"sonic\"\n\t\t}\n\t}\n}\n"
+
+		subprocess.call("cat <<EOF | sudo config reload /dev/stdin -y" + cfg_db_str, shell=True)
+		subprocess.call("sudo sonic-cfggen -H --write-to-db ", shell=True)
+		subprocess.call("sonic-cfggen -t /usr/local/lib/python2.7/dist-packages/usr/share/sonic/templates/l2switch.j2 -p -k" + l2_sku + " | sudo config load /dev/stdin -y", shell=True ) 
+		subprocess.call("sudo config save -y ", shell=True)
+		subprocess.call("sudo systemctl restart swss ", shell=True)
+
+		print( "Finished L2 mode configuration based on "+l2_sku+" SKU")
+		
+def main():
+	parser = argparse.ArgumentParser(description='Create a new SKU',
+									version='1.0.0',
+									formatter_class=argparse.RawTextHelpFormatter)
+	parser.add_argument('-f', '--file', action='store', help='SKU definition file (default <cwd>/sku_def.xml)', default=None)
+	parser.add_argument('-b', '--base', action='store', help='SKU base definition  ', default=None)
+	parser.add_argument('-r', '--remove', action='store_true', help='Remove SKU folder')
+	parser.add_argument('-c', '--cmd', action='store', choices=['new_sku_only', 'l2_mode_only', 'new_sku_l2'], help='Choose action to preform (Generate a new SKU, Configure L2 mode, Both', default="new_sku_only")
+	parser.add_argument('-l2_sku_name', '--l2_sku_name', action='store', help='SKU name to be used in the L2 configuration mode', default=None)
+	parser.add_argument('-p', '--print', action='store_true', help='Print port_config.ini without creating a new SKU', default=False)
+	parser.add_argument('-vv', '--verbose', action='store_true', help='Verbose output', default=False)
+	args = parser.parse_args()
+	
+	l2_mode = False
+	sku_mode = False
+	sku_name = None
+	try:
+		sku = SkuCreate()
+		if args.cmd == "l2_mode_only":
+			l2_mode = True
+			sku_mode = False
+		elif args.cmd == "new_sku_l2":
+			l2_mode = True
+			sku_mode = True
+		else :
+			l2_mode = False
+			sku_mode = True		
+		
+		if args.file:
+			SKU_DEF_FILE = args.file
+		else:
+			SKU_DEF_FILE = os.getcwd() + '/sku_def.xml'
```

Comment:
>os.getcwd() + '/sku_def.xml [](start = 18, length = 27)

The default path is based on current directory, which may be some random path user is reside in. Let's keep teh argument as required.

Reply:
Fixed. Verifying SKU definition file is provided when creating a new SKU.
When setting l2_mode_only, it is allowed to omit the file and the current SKU will be used